lec02-basic-r.Rmd

```diff
@@ -1,4 +1,765 @@
+lines you want to comment, press at the same time on your keyboard
+<kbd>Ctrl</kbd> + <kbd>Shift</kbd> + <kbd>C</kbd>. If you only want to comment
+out one line, you can put the cursor at any location of that line (i.e. no need 
+to select the whole line), then press <kbd>Ctrl</kbd> + <kbd>Shift</kbd> + 
+<kbd>C</kbd>.
+
+```{r}
+weight_lb <- 2.2 * weight_kg # Actually, 1 kg = 2.204623 lbs
+```
+
+> **Challenge**
+>
+> What are the values after each statement in the following?
+
+> ```{r}
+> mass <- 47.5            # mass?
+> age  <- 122             # age?
+> mass <- mass * 2.0      # mass?
+> age  <- age - 20        # age?
+> mass_index <- mass/age  # mass_index?
+> ```
+
+
+### Functions and their arguments
+
+Functions can be thought of as recipes. You give a few ingredients as input to a
+function, and it will generate an output based on these ingredients. Just as
+when baking, both the ingredients and the actual recipe will influence what
+comes out of the recipe in the end, will it be a cake or a loaf of bread? In R,
+the inputs to a function are not called ingredients, but rather *arguments*, and
+the output is called the *return value* of the function. A function does not
+technically have to return a value, but often does so. Functions are used to
+automate more complicated sets of commands and many of them are already
+predefined in R. A typical example would be the function `sqrt()`. The input
+(the argument) must be a number, and the return value (in fact, the output) is
+the square root of that number. Executing a function ('running it') is called
+*calling* the function. An example of a function call is:
+
+```{r}
+sqrt(9)
+```
+
+Which is the same as assigning the value to a variable and then passing that
+variable to the function:
+
+```{r}
+a <- 9
+b <- sqrt(a)
+b
+```
+
+Here, the value of `a` is given to the `sqrt()` function, the `sqrt()` function
+calculates the square root, and returns the value which is then assigned to
+variable `b`. This function is very simple, because it takes just one argument.
+
+The return 'value' of a function need not be numerical (like that of `sqrt()`),
+and it also does not need to be a single item: it can be a set of things, or
+even a dataset, as we will see later on.
+
+Arguments can be anything, not only numbers or filenames, but also other
+objects. Exactly what each argument means differs per function, and must be
+looked up in the documentation (see below). Some functions take arguments which
+may either be specified by the user, or, if left out, take on a *default* value:
+these are called *options*. Options are typically used to alter the way the
+function operates, such as whether it ignores 'bad values', or what symbol to
+use in a plot. However, if you want something specific, you can specify a value
+of your choice which will be used instead of the default.
+
+To access help about `sqrt`, we are first going to learn about tab-completion.
```

Comment:
Sort of already did this.

Reply:
But maybe it's still good to do it explicitly? It's an important thing, and if someone misses it it would be bad.